pkg/tnf/handlers/podantiaffinity/podantiaffinity_test.go

```diff
@@ -0,0 +1,144 @@
+	pathToTestSchemaFile  = path.Join(pathRelativeToRoot, genericTestSchemaFile)
+	testDeploymentName    = "testDeployName123"
+	testNamespace         = "testDeployNamespace123"
+)
+
+func createTest() (*tnf.Tester, []reel.Handler, *gojsonschema.Result, error) {
+	values := make(map[string]interface{})
+	values["DEPLOYMENT_NAME"] = testDeploymentName
+	values["DEPLOYMENT_NAMESPACE"] = testNamespace
+	return generic.NewGenericFromMap(checkSubFilename, pathToTestSchemaFile, values)
+}
+
+func TestPods_Args(t *testing.T) {
+	test, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Nil(t, (*test).Args())
+}
+
+func TestPods_GetIdentifier(t *testing.T) {
+	test, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, identifier.PodAntiAffinityIdentifier, (*test).GetIdentifier())
+}
+
+func TestPods_ReelFirst(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	step := handler.ReelFirst()
+	expectedReelFirstExecute := fmt.Sprintf(
+		"oc get deployment %s -n %s -o json | jq -r ' if .spec.replicas > 1 then "+
+			"if .spec.template.spec.affinity.podAntiAffinity != null then \"OK\" else "+
+			"\"Antiaffinity missing\" end else \"Replica count is 1\" end '\n",
+		testDeploymentName, testNamespace)
+	assert.Equal(t, expectedReelFirstExecute, step.Execute)
+	assert.Contains(t, step.Expect, expectedPassPattern, expectedFailPattern1, expectedFailPattern2)
+	assert.Equal(t, testTimeoutDuration, step.Timeout)
+}
+
+func TestPods_ReelEOF(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	// just ensure there isn't a panic
+	handler.ReelEOF()
+}
+
+func TestPods_ReelTimeout(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	assert.Nil(t, handler.ReelTimeout())
+}
+
+func TestPods_ReelMatch(t *testing.T) {
+	tester, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+
+	// Positive Test
+	step := handler.ReelMatch(expectedPassPattern, "", "OK")
+	assert.Nil(t, step)
+	assert.Equal(t, tnf.SUCCESS, (*tester).Result())
+
+	// Negative Test
+	step = handler.ReelMatch(expectedFailPattern1, "", "Antiaffinity missing")
+	assert.Nil(t, step)
+	assert.Equal(t, tnf.FAILURE, (*tester).Result())
```

Comment:
Just a comment, no need to change. Seems like for simple json tests, we could skip unit testing altogether? This testing does not seem very useful as we pass input and output directly

Reply:
The unit test might still be required to test the json file itself

Reply:
I had the same question the first time I saw this json based generic test handler implementation. Ryan answered my question in the last training video. I wasn't fully convinced. Need to look into the actual code to understand how to unit test it better.